src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -86,731 +128,12088 @@ namespace AirflowNetwork {
-        // allocate same size as others -- this will be maximum  !noel
-        newAU.allocate(IK(NetworkNumOfNodes + 1));
-#endif
+            calculate_heat_balance();
+            calculate_moisture_balance();
+            if (m_state.dataContaminantBalance->Contaminant.CO2Simulation) calculate_CO2_balance();
+            if (m_state.dataContaminantBalance->Contaminant.GenericContamSimulation) calculate_GC_balance();
+        }
 
-        // noel, GNU says the AU is indexed above its upper bound
-        // ALLOCATE(AU(IK(NetworkNumOfNodes+1)-1))
-        AU.allocate(IK(NetworkNumOfNodes + 1));
+        update(FirstHVACIteration);
     }
 
-    void Solver::initialize(EnergyPlusData &state)
+    bool Solver::get_element_input()
     {
-
         // SUBROUTINE INFORMATION:
-        //       AUTHOR         Lixing Gu
-        //       DATE WRITTEN   Aug. 2003
+        //       AUTHOR         Jason DeGraw
+        //       DATE WRITTEN   Oct. 2018
         //       MODIFIED       na
         //       RE-ENGINEERED  na
 
         // PURPOSE OF THIS SUBROUTINE:
-        // This subroutine initializes variables for AirflowNetworkSolver.
+        // This subroutine reads airflow element inputs (eventually)
+
+        static constexpr std::string_view RoutineName{"get_element_input"};
+        std::string CurrentModuleObject;
+        bool success{true};
+
+        // *** Read AirflowNetwork simulation reference crack conditions
+        std::unordered_map<std::string, ReferenceConditions> referenceConditions; // Map for lookups
+        ReferenceConditions defaultReferenceConditions("Default");                // Defaulted conditions
+        bool conditionsAreDefaulted(true);                                        // Conditions are defaulted?
+        CurrentModuleObject = "AirflowNetwork:MultiZone:ReferenceCrackConditions";
+        auto instances = m_state.dataInputProcessing->inputProcessor->epJSON.find(CurrentModuleObject);
+        if (instances != m_state.dataInputProcessing->inputProcessor->epJSON.end()) {
+            // globalSolverObject.referenceConditions.clear();
+            auto &instancesValue = instances.value();
+            for (auto instance = instancesValue.begin(); instance != instancesValue.end(); ++instance) {
+                auto const &fields = instance.value();
+                auto const &thisObjectName = UtilityRoutines::MakeUPPERCase(instance.key());
+                Real64 temperature(20.0);
+                if (fields.find("reference_temperature") != fields.end()) { // required field, has default value
+                    temperature = fields.at("reference_temperature").get<Real64>();
+                }
+                Real64 pressure(101325.0);
+                if (fields.find("reference_barometric_pressure") != fields.end()) { // not required field, has default value
+                    pressure = fields.at("reference_barometric_pressure").get<Real64>();
```

Comment:
There are new worker functions (#8738), would they not work here? or are more complicated than needed? I haven't used them yet so not sure.

    getAlphaFieldValue(state, CurrentModuleObject, objectFields, objectSchemaProps, "alpha_field_name")
    getRealFieldValue(state, CurrentModuleObject, objectFields, objectSchemaProps, "real_field_name")

    e.g., thisMaterial.Thickness = ip->getRealFieldValue(objectFields, objectSchemaProps, "thickness");

Reply:
I totally forgot about that!! Looking at that code, it looks like it handles defaults in an inefficient way, but maybe I just don't understand what the epJSON default semantics are. It looks like you get a default if the field doesn't appear in the object (okay) and also if the field appears in the object but it has no value (is this a thing? Is this only a thing when you are translating from IDF?). Default finding also does automatic type conversions, e.g., if you expect a string and you find an integer it writes that integer to a string, etc. Feels like a lot is going on there. It's trying to be too robust to handle the possibility of poorly formed input. But it should work. We can try to lean out that code also.

Reply:
This was originally done before the worker functions were a thing, and this object was a little strange to being with in that originally had no required fields except for the name. We'll be addressing the defaults and all that in the epJSON work that @mbadams5 is starting.

Reply:
Sounds like a go to me. @jasondegraw I wasn't suggesting this change, just pointing out the resource.